src/v/transform/transform_manager.cc

```diff
@@ -0,0 +1,386 @@
+          [](auto& e) { return e.processor()->stop(); });
+        by_ntp.erase(range.first, range.second);
+    }
+
+private:
+    underlying_t _underlying;
+};
+
+manager::manager(
+  std::unique_ptr<registry> r, std::unique_ptr<processor_factory> f)
+  : _queue([](const std::exception_ptr& ex) {
+      vlog(tlog.error, "unexpected transform manager error: {}", ex);
+  })
+  , _registry(std::move(r))
+  , _processors(std::make_unique<processor_table>())
+  , _processor_factory(std::move(f)) {}
+
+manager::~manager() = default;
+
+ss::future<> manager::start() { return ss::now(); }
+ss::future<> manager::stop() {
+    vlog(tlog.info, "Stopping transform manager...");
+    co_await _queue.shutdown();
+    co_await _processors->clear();
+    vlog(tlog.info, "Stopped transform manager.");
+}
+
+void manager::on_leadership_change(model::ntp ntp, ntp_leader leader_status) {
+    _queue.submit([this, ntp = std::move(ntp), leader_status]() mutable {
+        return handle_leadership_change(std::move(ntp), leader_status);
+    });
+}
+void manager::on_plugin_change(model::transform_id id) {
+    _queue.submit([this, id] { return handle_plugin_change(id); });
+}
+void manager::on_transform_error(
+  model::transform_id id, model::ntp ntp, model::transform_metadata meta) {
+    _queue.submit(
+      [this, id, ntp = std::move(ntp), meta = std::move(meta)]() mutable {
+          return handle_transform_error(id, std::move(ntp), std::move(meta));
+      });
+}
+
+ss::future<>
+manager::handle_leadership_change(model::ntp ntp, ntp_leader leader_status) {
+    vlog(
+      tlog.debug,
+      "handling leadership status change to leader={} for: {}",
+      leader_status,
+      ntp);
+
+    if (leader_status == ntp_leader::no) {
+        // We're not the leader anymore, time to shutdown all the processors
+        co_await _processors->erase_by_ntp(ntp);
+        co_return;
+    }
+    // We're the leader - start all the processor that aren't already running
+    auto transforms = _registry->lookup_by_input_topic(
+      model::topic_namespace_view(ntp));
+    co_await ss::parallel_for_each(
+      std::move(transforms),
+      [this, &ntp](auto entry) { return start_processor(ntp, entry.first); });
+}
+
+ss::future<> manager::handle_plugin_change(model::transform_id id) {
+    vlog(tlog.debug, "handling update to plugin: {}", id);
+    // If we have an existing processor we need to restart it with the updates
+    // applied.
+    co_await _processors->erase_by_id(id);
+
+    auto transform = _registry->lookup_by_id(id);
+    // If there is no transform we're good to go, everything is shutdown if
+    // needed.
+    if (!transform) {
+        co_return;
+    }
+    // Otherwise, start a processor for every partition we're a leader of.
+    auto partitions = _registry->get_leader_partitions(transform->input_topic);
+    co_await ss::parallel_for_each(
+      partitions.begin(),
+      partitions.end(),
+      [this, id, transform = *std::move(transform)](
```

Comment:
i would have expected `std::move(*transform)` but now i'm wondering if they end up being equivalent?

Reply:
clang-tidy reports those are the same

Reply:
fun